convert gother.c to D

WalterBright opened the pull request.

WalterBright committed:
convert gother.c to D

wilzbach commented:
We really need to upgrade gdc:

```
dmd/backend/global.d:71:17: error: variable SCMAX extern symbols cannot have initializers char[SCMAX] sytab;
```

wilzbach committed:
Add workaround for the gdc extern symbols with initializers build failure

dmd/backend/global.d:71:17: error: variable SCMAX extern symbols cannot have initializers char[SCMAX] sytab;

wilzbach commented:
Turns out you can workaround the gdc failure with:

```
enum SCMAX_ = SCMAX;
```

(I added this as a commit to check whether the CIs agree.)
